code/datums/components/beacon.dm

```diff
@@ -14,8 +14,10 @@
 	var/active_icon_state = ""
 	///The mob who activated this beacon
 	var/mob/activator
+	///Contains the item the beacon was innitially created with
+	var/obj/item/sourceitem
```

Comment:
This is worthless, `parent` already tells you what the component is attached to, there is no reason to have this. Also this is bad code because you're assuming the thing the beacon component is attached to will always be an item (it won't be, see AI drone subtype)

Reply:
Thanks